src/components/ArticleListByTopic.jsx

```diff
@@ -0,0 +1,34 @@
+import { useEffect, useState } from 'react';
+import { useParams, useSearchParams } from 'react-router-dom';
+import { getArticlesByTopic } from '../utils/api';
+import ArticleCard from './ArticleCard';
+import ArticleSearch from './ArticleSearch';
+import styles from '../styles/ArticleList.module.css';
+
+const ArticleListByTopic = () => {
```

Comment:
This looks good but seems very similar to articleList, could this be incorporated into the same component, keeos it D.R.Y.